crates/sui-framework/sources/collectible/transfer_policy.move

```diff
@@ -0,0 +1,448 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Part of the collectibles bundle - a primitive allowing creators to enforce
+/// constraints on transfers as long as the transfers are performed in the ecosystem
+/// that enforces them.
+module sui::transfer_policy {
+    use std::vector;
+    use std::option::{Self, Option};
+    use std::type_name::{Self, TypeName};
+    use sui::package::{Self, Publisher};
+    use sui::tx_context::TxContext;
+    use sui::object::{Self, ID, UID};
+    use sui::vec_set::{Self, VecSet};
+    use sui::dynamic_field as df;
+    use sui::bag::{Self, Bag};
+    use sui::balance::{Self, Balance};
+    use sui::sui::SUI;
+    use sui::coin::{Self, Coin};
+    use sui::event;
+
+    friend sui::collectible;
+
+    /// The number of receipts does not match the `TransferPolicy` requirement.
+    const EPolicyNotSatisfied: u64 = 0;
+    /// A completed rule is not set in the `TransferPolicy`.
+    const EIllegalRule: u64 = 1;
+    /// A Rule is not set.
+    const EUnknownRequrement: u64 = 2;
+    /// Attempting to create a Rule that is already set.
+    const ERuleAlreadySet: u64 = 3;
+    /// Trying to `withdraw` or `close_and_withdraw` with a wrong Cap.
+    const ENotOwner: u64 = 4;
+    /// Trying to `withdraw` more than there is.
+    const ENotEnough: u64 = 5;
+
+    /// A "Hot Potato" forcing the buyer to get a transfer permission
+    /// from the item type (`T`) owner on purchase attempt.
+    struct TransferRequest<phantom T: key + store> {
+        /// Amount of SUI paid for the item. Can be used to
```

Comment:
Does it have to be SUI or could people use this to manage payments for transfers in terms of other coins?

Reply:
Only SUI for now.